smartapps/tonesto7/nest-manager.src/nest-manager.groovy

```diff
@@ -346,93 +349,40 @@ def pollWatcher(evt) {
 }
 
 def poll(force = false, type = null) {
-	schedFollowPoll()
-   	if(isPollAllowed()) { 
-   		def dev = false
-        def str = false
-        if (force == true) { forcedPoll(type) }
+	if(isPollAllowed()) { 
+	    def dev = false
+	    def str = false
+	    if (force == true) { forcedPoll(type) }
 	    if ( !force && !ok2PollDevice() && !ok2PollStruct() ) {
-			LogAction("It's Too Soon to Refresh Data!!! - Devices Last Updated: ${getLastDevicePollSec()} seconds ago... | Structures Last Updated ${getLastStructPollSec()} seconds ago...", "info", true)
-            scheduleNextPoll()
- 		}
-   		else if(!force) {
-    		scheduleNextPoll()
-   			if(ok2PollDevice()) { 
-            	LogAction("Updating Device Data...(Last Updated: ${getLastDevicePollSec()} seconds ago)", "info", true)
-            	dev = getApiData("dev")
-            }
-            if(ok2PollStruct()) {
-            	LogAction("Updating Structure Data...(Last Updated: ${getLastStructPollSec()} seconds ago)", "info", true)
-                str = getApiData("str")
-            }    
-		}
-        if(updChildOnNewOnly) {
-        	if (dev || str || (getLastChildUpdSec() > 1800)) { updateChildData() }
-        } else { updateChildData() }
+		    LogAction("Nothing to do - Devices Last Updated: ${getLastDevicePollSec()} seconds ago... | Structures Last Updated ${getLastStructPollSec()} seconds ago...", "info", true)
+	    }
+	    else if(!force) {
+    		if(ok2PollStruct()) {
+		        LogAction("Updating Structure Data...(Last Updated: ${getLastStructPollSec()} seconds ago)", "info", true)
+		        str = getApiData("str")
+    		}    
+   		    if(ok2PollDevice()) { 
+		        LogAction("Updating Device Data...(Last Updated: ${getLastDevicePollSec()} seconds ago)", "info", true)
+		        dev = getApiData("dev")
+    	    }
+	    }
+	    if(updChildOnNewOnly) {
+        	if (dev || str || atomicState?.NeedChildUpd || (getLastChildUpdSec() > 1800)) { updateChildData() }
+	    } else { updateChildData() }
 	}
     if(getLastWebUpdSec() > 1800) {
-       	getWebFileData() //This reads a JSON file from a web server with timing values and version numbers
+        if(canSchedule()) { runIn(10, "getWebFileData", [overwrite: true]) }  //This reads a JSON file from a web server with timing values and version numbers
```

Comment:
It was causing timeouts some times, so I decide to control it better.

On Apr 2, 2016, at 1:43 PM, Tonesto7 <notifications@github.com<mailto:notifications@github.com>> wrote:


In smartapps/tonesto7/nest-manager.src/nest-manager.groovy<https://github.com/tonesto7/nest-manager/pull/63#discussion_r58296944>:

>       }
>      if(getLastWebUpdSec() > 1800) {
> -             getWebFileData() //This reads a JSON file from a web server with timing values and version numbers
> +        if(canSchedule()) { runIn(10, "getWebFileData", [overwrite: true]) }  //This reads a JSON file from a web server with timing values and version numbers


What made you move this into a scheduled function?

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/tonesto7/nest-manager/pull/63/files/5f58a323e635a5abbd1b694c9b4212b9caced9b2#r58296944>